### DISCUSSION TITLE
ProjectionExpression for collections

### DISCUSSION THREAD

**Discussion by robkallen** [tywalch/electrodb#138] [General]
@tywalch first of all, great work on this project.  I noticed you recently added support for ProjectionExpression attributes for `get` and `query` operations, however this does not appear to be supported when using Service collections.  Is this by design, or are you planning to add support for this in the future?  Thanks!

**Comment by tywalch** [comment]
> @tywalch first of all, great work on this project.  I noticed you recently added support for ProjectionExpression attributes for `get` and `query` operations, however this does not appear to be supported when using Service collections.  Is this by design, or are you planning to add support for this in the future?  Thanks!

@robkallen I do plan to support collections soon in the next release. It wasn't include in the first one to prioritize getting it out the door. ProjectionExpressions in collection queries and scans coming soon 👍

**Reply by tywalch** [reply]
I'll add one more thing, with many-to-many you'll likely use multiple requests to solve the access pattern. That said, it is worth considering how you might model your schema in a denormalized form. Obviously it depends on your data, and it can be a paradigm shift, but it can go a long way to keep speeds fast and cost low.

**Comment by tywalch** [comment]
@robkallen as a small update a change coming in the next week or so will lay the groundwork for this. I also plan on adding a "hide" array which will act as a narrowing property list (the opposite of "attributes")
